lib/css/CssModulesPlugin.js

```diff
@@ -826,25 +823,108 @@ class CssModulesPlugin {
-					module,
-					runtimeTemplate,
-					hooks
-				});
-				source.add(moduleSource);
+				const [moduleSource, moduleExport, moduleIcssImport] =
+					this.renderModule({
+						metaData,
+						undoPath,
+						chunk,
+						chunkGraph,
+						codeGenerationResults,
+						module,
+						runtimeTemplate,
+						hooks
+					});
+				if (moduleSource) {
+					moduleSources.set(module, moduleSource);
+				}
+				if (moduleExport) {
+					moduleExports.set(module, moduleExport);
+				}
+				if (moduleIcssImport) {
+					moduleIcssImports.set(module, moduleIcssImport);
+				}
 			} catch (err) {
 				/** @type {Error} */
 				(err).message += `\nduring rendering of css ${module.identifier()}`;
 				throw err;
 			}
 		}
+		for (const [module, moduleSource] of moduleSources.entries()) {
+			let replaceSource = String(moduleSource.source());
+
+			const icssImports = moduleIcssImports.get(module);
+			if (icssImports) {
+				for (const importName of Object.keys(icssImports)) {
+					const resolvedModule = icssImports[importName].module;
+					let moduleId = chunkGraph.getModuleId(resolvedModule);
+					if (typeof moduleId === "string") {
+						moduleId = moduleId.replace(/\\/g, "/");
+					}
+					// get resolvedModule's exports
+					const resolvedExports = moduleExports.get(resolvedModule);
+					const importValue = icssImports[importName].importValue;
+					if (resolvedExports) {
+						replaceSource = replaceSource.replaceAll(
+							CssIcssImportDependency.generateReplacementStr(
+								String(moduleId),
+								importValue
+							),
+							resolvedExports.exports.get(importValue)
+						);
+					}
+				}
+			}
+			// icss re-export
+			const exportsData = moduleExports.get(module);
+			if (exportsData && icssImports) {
+				for (const [exportName, exportValue] of exportsData.exports.entries()) {
+					for (const token of Object.keys(icssImports)) {
+						if (token === exportValue) {
+							const reExportValue = icssImports[token].importValue;
+							const resolvedModule = icssImports[token].module;
+							// get resolvedModule's exports
+							const resolvedExports = moduleExports.get(resolvedModule);
+							if (resolvedExports) {
+								exportsData.exports.set(
+									exportName,
+									resolvedExports.exports.get(reExportValue)
+								);
+							}
+						}
+					}
+				}
+			}
+			const esModule = exportsData && exportsData.esModule;
+			const exports = exportsData && exportsData.exports;
+			let moduleId = String(chunkGraph.getModuleId(module));
+
+			// When `optimization.moduleIds` is `named` the module id is a path, so we need to normalize it between platforms
+			if (typeof moduleId === "string") {
+				moduleId = moduleId.replace(/\\/g, "/");
+			}
+
+			metaData.push(
+				`${
+					exports
+						? Array.from(
+								exports,
+								([n, v]) => `${escapeCss(n)}:${escapeCss(v)}/`
+							).join("")
+						: ""
+				}${esModule ? "&" : ""}${escapeCss(moduleId)}`
+			);
+
+			source.add(replaceSource);
+		}
```

Comment:
It is very slow, we need to search another approach

Reply:
Maybe we can define  a CSS variable for each CSS export, and each ICSS import will be replaced by the CSS variable

Reply:
except for cases when they are really needed

Reply:
I will try again to collect replacementRange and then replace them during renderChunk. However, this requires ensuring that the source remains as replaceSource from the generation phase before applying replacements. Currently, `InitFragment.addToSource` in generate and renderModule both modify the source, so I plan to move these operations after replacing the range.